### ISSUE TITLE
Cookies are not correctly returned from getAllCookies

### ISSUE THREAD

**Issue by chrissound** [nbloomf/webdriver-w3c#41]
`getAllCookies` is not responding with the appropriate cookie values - which are present if I inspect firefox's developer tools. This is using the latest tar from hackage (webdriver-w3c-0.0.2).

```
{-# LANGUAGE OverloadedStrings #-}


module Main where

import Web.Api.WebDriver.Endpoints
import Web.Api.WebDriver
import Control.Monad.IO.Class
import Test.Tasty
import Test.Tasty.WebDriver


main :: IO ()
main = do
  defaultWebDriverMain  $ testGroup "Test Demo"
    $ [ testCase "testXyz" testXyz]

liftwd :: IO a -> WebDriverT IO a
liftwd = liftWebDriverTT . liftIO


testXyz :: WebDriverT IO ()
testXyz = do

  fullscreenWindow
  navigateTo "https://httpbin.org/cookies/set?freeform=abc"

  wait 2000000

  c' <- getAllCookies

  assertNotEqual [] c' ( "Cookies should be present")

  liftwd $ do
    print (_cookieName <$> c')
    print (_cookieValue <$> c')
```
https://httpbin.org/cookies/set?freeform=abc is a little web debug helper to set a cookie to demonstrate this issue. 

This seems to only occur with the tasty testing. Meanwhile the following works perfectly: 
```
{-# LANGUAGE OverloadedStrings #-}


module Main where

import Web.Api.WebDriver.Endpoints
import Web.Api.WebDriver
import Control.Monad.IO.Class

main :: IO ()
main = do
  x <- execWebDriverT defaultWebDriverConfig $
   runIsolated defaultFirefoxCapabilities doLogin
  print x

liftwd :: IO a -> WebDriverT IO a
liftwd = liftWebDriverTT . liftIO


doLogin :: WebDriverT IO ()
doLogin = do
  fullscreenWindow
  navigateTo "https://httpbin.org/cookies/set?freeform=abc"
  c' <- getAllCookies
  assertNotEqual [] c' ( "Cookies should be present")
```

**Comment by chrissound** [comment]
Perhaps a bug in the geckodriver? As the logs seem to show it returning an empty set for cookies... Unless there is something that the webdriver client needs to set? 

```
>>> Deployment environment is DEV
>>> Logging with colors
Test Demo
2021-12-13 18:29:08 INFO  fa3c5997-1 localhost:4444 Comment testXyz
2021-12-13 18:29:08 INFO  fa3c5997-1 localhost:4444 Comment Attempt #1
2021-12-13 18:29:08 DEBUG  fa3c5997-1 localhost:4444 Request Request POST http://localhost:4444/session
{
    "capabilities": {
        "alwaysMatch": {
            "moz:firefoxOptions": {
                "log": {
                    "level": "info"
                },
                "args": [
                    "-private"
                ]
            },
            "browserName": "firefox"
        }
    },
    "desiredCapabilities": {
        "moz:firefoxOptions": {
            "log": {
                "level": "info"
            },
            "args": [
                "-private"
            ]
        },
        "browserName": "firefox"
    }
}
2021-12-13 18:29:10 DEBUG  fa3c5997-1 localhost:4444 Response Response
{
    "value": {
        "sessionId": "131583de-5d7f-49ef-a6a2-50abad6fe9af",
        "capabilities": {
            "moz:webdriverClick": true,
            "unhandledPromptBehavior": "dismiss and notify",
            "platformVersion": "5.4.122",
            "moz:headless": false,
            "moz:useNonSpecCompliantPointerOrigin": false,
            "moz:geckodriverVersion": "0.29.1",
            "setWindowRect": true,
            "browserVersion": "89.0.2",
            "pageLoadStrategy": "normal",
            "moz:profile": "/tmp/rust_mozprofileO5Uc28",
            "moz:accessibilityChecks": false,
            "moz:buildID": "20210630090118",
            "moz:processID": 8094,
            "strictFileInteractability": false,
            "platformName": "linux",
            "timeouts": {
                "implicit": 0,
                "script": 30000,
                "pageLoad": 300000
            },
            "acceptInsecureCerts": false,
            "browserName": "firefox",
            "moz:shutdownTimeout": 60000
        }
    }
}
2021-12-13 18:29:10 DEBUG  fa3c5997-1 localhost:4444 Request Request POST http://localhost:4444/session/131583de-5d7f-49ef-a6a2-50abad6fe9af/window/fullscreen
{}
2021-12-13 18:29:15 DEBUG  fa3c5997-1 localhost:4444 Response Response
{
    "value": {
        "height": 273,
        "width": 420,
        "x": 2131,
        "y": 886
    }
}
2021-12-13 18:29:15 DEBUG  fa3c5997-1 localhost:4444 Request Request POST http://localhost:4444/session/131583de-5d7f-49ef-a6a2-50abad6fe9af/url
{
    "url": "https://httpbin.org/cookies/set?freeform=abc"
}
2021-12-13 18:29:16 DEBUG  fa3c5997-1 localhost:4444 Response Response
{
    "value": null
}
2021-12-13 18:29:16 INFO  fa3c5997-1 localhost:4444 Pause Wait for 2000000μs
2021-12-13 18:29:18 DEBUG  fa3c5997-1 localhost:4444 Request Request POST http://localhost:4444/session/131583de-5d7f-49ef-a6a2-50abad6fe9af/url
{
    "url": "https://httpbin.org/cookies"
}
2021-12-13 18:29:18 DEBUG  fa3c5997-1 localhost:4444 Response Response
{
    "value": null
}
2021-12-13 18:29:18 DEBUG  fa3c5997-1 localhost:4444 Request Request GET http://localhost:4444/session/131583de-5d7f-49ef-a6a2-50abad6fe9af/cookie
2021-12-13 18:29:18 DEBUG  fa3c5997-1 localhost:4444 Response Response
{
    "value": []
}
2021-12-13 18:29:18 NOTICE  fa3c5997-1 localhost:4444 Log LogAssertion (Assertion {assertionStatement = [] is not equal to [], assertionComment = Cookies should be present, assertionResult = AssertFailure})
  testXyz: []
[]
2021-12-13 18:29:18 DEBUG  fa3c5997-1 localhost:4444 Request Request DELETE http://localhost:4444/session/131583de-5d7f-49ef-a6a2-50abad6fe9af
2021-12-13 18:29:18 DEBUG  fa3c5997-1 localhost:4444 Response Response
{
    "value": null
}
FAIL (10.25s)
    Invalid Assertion 
    assertion: [] is not equal to [] 
    comment: Cookies should be present

1 out of 1 tests failed (10.25s)
*** Exception: ExitFailure 1
```

**Comment by nbloomf** [comment]
Thank you for this report! I think this is happening because the tasty integration uses an option (`wd-private-mode`) to enable private mode for the session, and the option is set to `True` by default. This is a bug, and doesn't even match the documentation for that option. I'll get a fix up shortly.

https://github.com/nbloomf/webdriver-w3c/blob/19e4db5d4132d5e48b50e8fc514d3dcb75acee37/src/Test/Tasty/WebDriver.hs#L493

**Comment by nbloomf** [comment]
Fixed by #46.
